plugins/BEdita/Core/src/Model/Table/ObjectsTable.php

```diff
@@ -151,10 +153,30 @@ public function beforeSave(Event $event, EntityInterface $entity)
             // Cannot save objects of an abstract type.
             return false;
         }
+        $this->checkLangTag($entity);
 
         return true;
     }
 
+    /**
+     * Check `lang` tag using `I18n` configuration.
+     *
+     * @param \Cake\Datasource\EntityInterface $entity Entity being saved.
+     * @return void
+     * @throws \Cake\Network\Exception\BadRequestException If a wrong lang tag is specified
+     */
+    protected function checkLangTag(EntityInterface $entity)
+    {
+        $defaultConf = ['languages' => [], 'default' => null];
```

Comment:
Setting the default language should definitely be here. I'd have done it slightly differently, though:
```php
if ($entity->isDirty('lang') && empty($entity['lang']) && Configure::check('I18n.default')) {
    $entity->set('lang', Configure::read('I18n.default'));
}
```

This way, we would set the default only if the `lang` field has been changed somehow. Otherwise it might happen that some old object has the language set when it's receiving modifications of some other nature.

-------------

Also, I'd prefer to move the language validation part in a custom validator. We might add a method `\BEdita\Core\Model\Validation\Validation::languageTag()` that looks more or less like this:
```php
public static function languageTag($tag)
{
    $languages = Hash::normalize((array)Configure::read('I18n.languages'));
    if (!empty($languages)) {
        return array_key_exists($tag, $languages);
    }

    // validate using some evil regex
}
```

Then, in `\BEdita\Core\Model\Validation\ObjectsValidator::__construct()`:
```php
$this
    // ...
    ->scalar('lang')
    ->add('lang', 'languageTag', ['rule' => [Validation::class, 'languageTag']])
    ->allowEmpty('lang')
```

Reply:
Done.
I've added Also a `I18nTest` integration test.